### PR TITLE
fill decode/encode routines for many messages with fuzz targets

### PR DESCRIPTION
**Pull request by yuntai**

### CODE REVIEW

**Comment by yuntai** [comment]
with rust-bitcoin/rust-secp256k1 tests fail.

File: Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ fuzztarget = ["secp256k1/fuzztarget", "bitcoin/fuzztarget"]
 bitcoin = "0.13"
 rust-crypto = "0.2"
 rand = "0.4"
-secp256k1 = "0.9"
+secp256k1 = { git = "https://github.com/rust-bitcoin/rust-secp256k1", branch = "master" , features=["fuzztarget"]}
```
Comment (TheBlueMatt) [inline Cargo.toml]: Shouldnt add the fuzztarget feature to all builds, thats why the tests are failing.

File: fuzz/fuzz_targets/msg_accept_channel_target.rs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+pub fn slice_to_be16(v: &[u8]) -> u16 {
+	((v[0] as u16) << 8*1) |
+	((v[1] as u16) << 8*0)
+}
+
+struct InputData<'a> {
+	data: &'a [u8],
+	read_pos: AtomicUsize,
+}
+impl<'a> InputData<'a> {
```
Comment (TheBlueMatt) [inline fuzz/fuzz_targets/msg_accept_channel_target.rs]: Shouldnt need all this complexity for message fuzz targets, check out how peer_crypt_target does it.

Comment (yuntai) [reply fuzz/fuzz_targets/msg_accept_channel_target.rs]: In channel_target.rs, I saw decode_msg_with_len16() & get_slice_nonadvancing() are used for a message with an optional field so thought it is better for fuzz to converge fast with that way?

Comment (TheBlueMatt) [reply fuzz/fuzz_targets/msg_accept_channel_target.rs]: Ah, sure, yea, but you can also add that without using a reader struct that gets passed around, significantly cutting down on code here.

Comment (yuntai) [reply fuzz/fuzz_targets/msg_accept_channel_target.rs]: ah, then I don't think I understood correctly how fuzz data feed works. you added Atomic for a multi-threaded test case while there should be no contention for `data` in `fuzz!(|data| {.. }` in a simple function, of course?

**Comment by TheBlueMatt** [comment]
Looks good, though can you swap out explicit lengths in the fuzz target for read lenghts? So instead of getting a slice of the right side, read 2 bytes for the length, then read that many bytes, try to deserialize it, and if it succeeds check that the serialized version is the beginning of the original.

**Comment by yuntai** [comment]
got it. but would like to know the rationale, please?

**Comment by TheBlueMatt** [comment]
a) it tests that the constants in decode are correct and that you can't get an out-of-bounds read, b) having fewer constants in the tests is good as it avoids any potential for a constant to be wrong resulting in in the test never successfully deserializing a message.

**Comment by yuntai** [comment]
thanks!

**Pull request merged**